dialer.go

```diff
@@ -153,16 +188,59 @@ func WithConnectionProperties(props amqp.Table) Option {
 	}
 }
 
+func WithReadyChan(receiver chan struct{}) Option {
+	f := func(c *Dialer) {
+		c.mu.Lock()
+		defer c.mu.Unlock()
+		c.readyChs = append(c.readyChs, receiver)
+	}
+	return f
+}
+
+func WithUnreadyChan(chUnready chan error) Option {
+	f := func(c *Dialer) {
+		c.mu.Lock()
+		defer c.mu.Unlock()
+		c.unreadyChs = append(c.unreadyChs, chUnready)
+	}
+	return f
+}
+
 func (c *Dialer) ConnectionCh() <-chan *Connection {
 	return c.connCh
 }
 
-func (c *Dialer) NotifyReady() <-chan struct{} {
-	return c.readyCh
+func (c *Dialer) NotifyReady(chReady chan struct{}) <-chan struct{} {
+	if cap(chReady) == 0 {
+		panic(errMsgReadyChanUnbuffered)
+	}
+
+	select {
+	case <- c.NotifyClosed():
+		return chReady
+	default:
+		c.mu.Lock()
+		defer c.mu.Unlock()
+		c.readyChs = append(c.readyChs, chReady)
+		return chReady
+	}
 }
 
-func (c *Dialer) NotifyUnready() <-chan error {
-	return c.unreadyCh
+func (c *Dialer) NotifyUnready(chUnready chan error) <-chan error {
+	if cap(chUnready) == 0 {
+		panic(errMsgUnreadyChanUnbuffered)
+	}
+
+	select {
+	case <- c.NotifyClosed():
+		close(chUnready)
+	default:
+		c.mu.Lock()
+		defer c.mu.Unlock()
+		c.unreadyChs = append(c.unreadyChs, chUnready)
+	}
+
+	return chUnready
```

Comment:
move return to `case <- c.NotifyClosed():`

Reply:
Left as it is